src/proof_rules/mciver_ast.rs

```diff
@@ -115,6 +117,15 @@ impl Encoding for ASTAnnotation {
         check_annotation_call(tycheck, call_span, &self.0, args)?;
         Ok(())
     }
+    fn check_calculus(&self, calculus: &Calculus, direction: Direction) -> Result<(), ()> {
+        if let CalculusType::WP = calculus.calculus_type {
```

Comment:
Why not just `if calculus.calculus_type == CalculusType::Wp && direction == Direction::Down`?